app/lib/hesa/client.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Hesa
+  class Client
+    AUTH_URL = "https://identity.hesa.ac.uk/Account/RemoteLogOn?ReturnDomain=https://datacollection.hesa.ac.uk&ReturnUrl=%2f"
+
+    class LoginError < StandardError; end
+
+    class UnauthorizedError < StandardError; end
+
+    def login
+      page = agent.get(AUTH_URL)
+      form = page.form_with(id: "loginForm")
+      form.EmailAddress = Settings.hesa.username
+      form.Password = Settings.hesa.password
+      page = form.submit
+
+      unless logged_in?(page)
+        raise(LoginError, "Unable to login to HESA with form")
+      end
+
+      page
+    end
+
+    def logged_in?(page)
+      !!(page.css("a").text.match("Log off") && page.css("a").text.match(Settings.hesa.username))
+    end
+
+    def agent
+      @agent ||= Mechanize.new
+    end
+
+    def get(url:)
+      login
+      page = agent.get(url)
+      page.body
+    rescue Mechanize::UnauthorizedError => e
```

Comment:
I'm not sure they're returning the right codes for this. I get a 302 and a 200. I was going to suggest loading an XmlDocument here and checking the root is <ITTRecord> but I think we might end up using this client for the other endpoints. Maybe we should just load an XmlDoc and if that succeeds return that otherwise error. Let the calling code decide whether it's the right XML for whatever request it is?